### DISCUSSION TITLE
testing if something is a quoted symbol

### DISCUSSION THREAD

**Discussion by DanielAjoy** [h3rald/min#183] [General]
Is there native operator to test if the element at the top of the stack is a quoted symbol? 

Is this a good way to do it?

> (((stringlike?) (string? not)) cleave and) ^quoted?

> 'a quoted?
> true

> (a) quoted?
> true

> ('a) quoted?
>  true

> $ 1 quoted?
>  false

> "a" quoted?
>  false

> "a" quotesym quoted?
> true

**Comment by h3rald** [comment]
Yup that works and that's a very concatenative way to do it 😊👍

The only minor problem is that the `and` symbol does not short-circuit; an alternative version (which defines a support symbol like an internal variable in non-concatenative languages would be:

```
(:a ((a stringlike?) (a string? not)) &&) ^quoted?
```
